### PR TITLE
add xpu pool3d kernels

### PR DESCRIPTION
**Pull request by ykkk2333**
### PR types

Others 

### PR changes

OPs 
### Describe

add xpu pool3d adn pool3d grad kernels,
fix xpu log grad bug, when operation is is inplace
xpu batch norm and batch norm grad kernel support NCDHW data layout, but the kernels is same as NCHW
delete xpu wait in adam kernel
reduce sum kernel supports fp16 input

### CODE REVIEW

File: paddle/phi/kernels/xpu/activation_grad_kernel.cc [hunk 1]
```diff
@@ -189,8 +189,10 @@ struct XPULogGradFunctor : public funcs::BaseActivationFunctor<T> {
     if (dOut != nullptr) dout_data = dOut->data<T>();
 
     T* dx_data = dev_ctx.template Alloc<T>(dX);
+    xpu::ctx_guard RAII_GUARD(dev_ctx.x_context());
+    float* tmp = RAII_GUARD.alloc_l3_or_gm<float>(x->numel());
```
Comment (QingshuChen) [inline paddle/phi/kernels/xpu/activation_grad_kernel.cc]: 这边分配应该是T类型？

File: paddle/phi/kernels/xpu/activation_grad_kernel.cc [hunk 2]
```diff
@@ -199,18 +201,16 @@ struct XPULogGradFunctor : public funcs::BaseActivationFunctor<T> {
...
                            reinterpret_cast<const float*>(x_data),
-                           reinterpret_cast<float*>(dx_data),
+                           reinterpret_cast<float*>(tmp),
                            x_dims,
                            x_dims);
     PADDLE_ENFORCE_XDNN_SUCCESS(r, "broadcast_div");
-
     r = xpu::broadcast_mul(dev_ctx.x_context(),
-                           reinterpret_cast<const float*>(dx_data),
+                           reinterpret_cast<const float*>(tmp),
```
Comment (QingshuChen) [inline paddle/phi/kernels/xpu/activation_grad_kernel.cc:213]: tmp的数据不需要memcpy到dx_data?

File: paddle/phi/kernels/xpu/pool_kernel.cc [hunk 3]
```diff
@@ -34,6 +34,8 @@ void Pool2dKernel(const Context& ctx,
                   const std::string& padding_algorithm,
                   DenseTensor* out) {
   using XPUType = typename XPUTypeTrait<T>::Type;
+  const bool channel_last = data_format == "NHWC";
```
Comment (houj04) [inline paddle/phi/kernels/xpu/pool_kernel.cc]: 这里增加了对其它data_format的支持之后，对应的单测也需要加上吧？

File: paddle/phi/kernels/xpu/pool_grad_kernel.cc [hunk 4]
```diff
@@ -95,51 +95,236 @@ void Pool2dGradKernel(const Context& ctx,
...
+
+  std::vector<int> paddings(paddings_t);
+  std::vector<int> kernel_size(kernel_size_t);
+  std::vector<int> strides(strides_t);
+
+  PADDLE_ENFORCE_EQ(
+      data_format,
+      "NCDHW",
+      phi::errors::InvalidArgument("The Pool3d_grad XPU OP only support"
+                                   "data_format is 'NCHW', but received %s",
```
Comment (houj04) [inline paddle/phi/kernels/xpu/pool_grad_kernel.cc]: NCHW -> NCDHW？

**Review by taixiurong — COMMENTED** [review]
LGTM

**Review by QingshuChen — APPROVED** [review]

**Pull request merged**